pkgs/os-specific/linux/dcgm/default.nix

```diff
@@ -0,0 +1,108 @@
+{ lib
+, callPackage
+, gcc11Stdenv
+, fetchFromGitHub
+, addOpenGLRunpath
+, catch2
+, cmake
+, cudaPackages_10_2
+, cudaPackages_11_8
+, cudaPackages_12
+, fmt_9
+, git
+, jsoncpp
+, libevent
+, plog
+, python3
+, tclap_1_4
+, yaml-cpp
+}:
+let
+  # Flags copied from DCGM's libevent build script
+  libevent-nossl = libevent.override { sslSupport = false; };
+  libevent-nossl-static = libevent-nossl.overrideAttrs (super: {
+    CFLAGS = "-Wno-cast-function-type -Wno-implicit-fallthrough -fPIC";
+    CXXFLAGS = "-Wno-cast-function-type -Wno-implicit-fallthrough -fPIC";
+    configureFlags = super.configureFlags ++ [ "--disable-shared" "--with-pic" ];
+  });
+
+  jsoncpp-static = jsoncpp.override { enableStatic = true; };
+
+  cudaToolkits = [
+    {
+      version = "10";
+      toolkit = cudaPackages_10_2.cudatoolkit;
+    }
+    {
+      version = "11";
+      toolkit = cudaPackages_11_8.cudatoolkit;
+    }
+    {
+      version = "12";
+      toolkit = cudaPackages_12.cudatoolkit;
+    }
+  ];
```

Comment:
Can you make some specific recommendations of how I would avoid this? [Here is how DCGM gets cudatoolkit](https://github.com/NVIDIA/DCGM/blob/7e1012302679e4bb7496483b32dcffb56e528c92/dcgmbuild/scripts/0080_cuda.sh).

Reply:
https://github.com/NVIDIA/DCGM/blob/7e1012302679e4bb7496483b32dcffb56e528c92/dcgmbuild/scripts/0080_cuda.sh#L24-L39

indicates what redist packages you're looking for. you can find the whole list of nix packages available with some tab completion-fu: `nix-build -A cudaPackages.<TAB>`

then it's just a matter of picking the ones you need. you can check https://github.com/NixOS/nixpkgs/pull/217410 for an example migration `cudatoolkit -> redist packages`

Reply:
Thank you. I will give that a shot.

Reply:
I was able to get this working for cudaPackages_11 and cudaPackages_12. Sadly, cudaPackages_10 does not have redist packages broken out at this level, it seems. I've made the dependencies granular for 11 and 12, and left 10 as `cudatoolkit`.

Reply:
I should point out that the result of my work is clearly much less clean than the previous solution because DCGM's CMake assumes all headers in the same tree structure, but if `cudatoolkit` is eventually being deprecated, then it might be worth the added complexity.

Reply:
Yes, and this additional complexity will be alleviated shortly with https://github.com/NixOS/nixpkgs/pull/233581

Reply:
Amazing. Little did I know I am not the first to resort to symlinkJoin 😂